app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -299,4 +359,125 @@ class ExplorationActivityPresenter @Inject constructor(
       ) as ExplorationFragment
     explorationFragment.revealSolution()
   }
+
+  private fun showProgressDatabaseFullDialogFragment() {
+    val previousFragment = activity.supportFragmentManager.findFragmentByTag(
+      TAG_PROGRESS_DATABASE_FULL_DIALOG
+    )
+    if (previousFragment != null) {
+      activity.supportFragmentManager.beginTransaction().remove(previousFragment).commitNow()
+    }
+
+    // If any one of oldestExplorationId or oldestExplorationTitle is not initialized, it means that
+    // there was an error while retrieving the oldest saved checkpoint details. In that the
+    // exploration is exited without deleting the any checkpoints.
+    if (
+      !::oldestCheckpointExplorationId.isInitialized ||
+      !::oldestCheckpointExplorationTitle.isInitialized
+    ) {
+      stopExploration()
+      return
+    }
+
+    val dialogFragment =
+      ProgressDatabaseFullDialogFragment.newInstance(oldestCheckpointExplorationTitle)
+    dialogFragment.showNow(
+      activity.supportFragmentManager,
+      TAG_PROGRESS_DATABASE_FULL_DIALOG
+    )
+  }
+
+  private fun showStopExplorationDialogFragment() {
+    val previousFragment =
+      activity.supportFragmentManager.findFragmentByTag(TAG_STOP_EXPLORATION_DIALOG)
+    if (previousFragment != null) {
+      activity.supportFragmentManager.beginTransaction().remove(previousFragment).commitNow()
+    }
+    val dialogFragment = StopExplorationDialogFragment.newInstance()
+    dialogFragment.showNow(activity.supportFragmentManager, TAG_STOP_EXPLORATION_DIALOG)
+  }
+
+  private fun showUnsavedExplorationDialogFragment() {
+    val previousFragment =
+      activity.supportFragmentManager.findFragmentByTag(TAG_UNSAVED_EXPLORATION_DIALOG)
+    if (previousFragment != null) {
+      activity.supportFragmentManager.beginTransaction().remove(previousFragment).commitNow()
+    }
+    val dialogFragment = UnsavedExplorationDialogFragment.newInstance()
+    dialogFragment.showNow(
+      activity.supportFragmentManager,
+      TAG_UNSAVED_EXPLORATION_DIALOG
+    )
+  }
+
+  /**
+   * This function listens to the result of [ExplorationDataController.getExplorationById].
```

Comment:
Function KDocs shouldn't start with "This function...". Instead, start with the verb.

Ditto everywhere in this PR.

Reply:
done